refinery/data_set_manager/test_tasks.py

```diff
@@ -0,0 +1,32 @@
+import os
+from data_set_manager.tasks import generate_bam_index
+from django.test import TestCase
+from file_store.models import FileType, FileStoreItem
+
+
+class BamFileIndexTest(TestCase):
+
+    def setUp(self):
+        self.file_store_item = FileStoreItem(
+            filetype=FileType.objects.get(name='BAM')
+        )
+        self.file_store_item.save()
+        self.folder = 'data_set_manager/test-data/'
+        self.test_bam_file = self.folder + 'test.bam'
+
+    def tearDown(self):
+        os.remove(self.test_bam_file + '.bai')
+
+    def test_index_bam_success_model_update(self):
+        generate_bam_index(self.file_store_item.uuid, self.test_bam_file)
+        self.assertEquals(
+            FileStoreItem.objects.get(
+                uuid=self.file_store_item.uuid).source,
+            'data_set_manager/test-data/test.bam.bai'
+        )
+
+    def test_index_bam_success_file_creation(self):
+        generate_bam_index(self.file_store_item.uuid, self.test_bam_file)
+        self.assertTrue(
+            os.path.exists(self.test_bam_file + '.bai')
+        )
```

Comment:
This is unnecessary because code under test is broken: (#2713 and #2718).

Also, for future reference:
* this is not a unit test (it requires a file system) and we've decided to keep those separate from integration/functional tests (#2539)
* we don't need to test 3rd party code (pysam) - mocking those calls should be used instead

Reply:
I was aware at the time I wrote this that it was not a unit test - I wanted nonetheless to make sure this function was working through the upgrade of it/python, especially since its documentation is not great.  Should I get rid of the test all together or keep it and mock the pysam function (which would make it unit but also then not give me the test of functionality I intended)?

Reply:
I would remove the test because the code under test is broken and will require a significant rewrite.